build-system/compile/closure-compile.js

```diff
@@ -69,10 +71,14 @@ function handleClosureCompilerError(err, outputFilename, options) {
  * on Windows exceeds the command line size limit. The stream mode is 'IN'
  * because output files and sourcemaps are written directly to disk.
  * @param {Array<string>} flags
+ * @param {!Object} options
  * @return {!Object}
  */
-function initializeClosure(flags) {
-  const pluginOptions = {streamMode: 'IN', logger: logClosureCompilerError};
+function initializeClosure(flags, options) {
+  const pluginOptions = {
+    streamMode: 'IN',
+    logger: options.logger || logClosureCompilerError,
+  };
```

Comment:
Suggestion: minimal args rather than full options object
```suggestion
function initializeClosure(flags, logger = logClosureCompilerError) {
  const pluginOptions = {streamMode: 'IN', logger};
```

Reply:
I wrote it this way initially, but most of the code throughout uses the entire options object. would rather conform with the rest of the files for now

Reply:
I think passing around an entire options object (especially given that it's an untyped one) when the function only consumes one property seems like an anti-pattern to me. Since this helper has nothing to do with the actual options, I feel like conformance for the sake of it isn't worthwhile here

Reply:
Would adding types to the options object that is passed everywhere alleviate your concerns with it?

Reply:
It would be good to do but it's not the main conflict. I think when helpers only use a couple values from an options object, it should be preferable to provide those values directly to keep logical concerns separate. Options objects are global-ish whereas making this function just take a logger makes it easier to refactor down the line if we wanted to pull it out for some other purpose. Though I see it's already been submitted so oh well.